Readme improvement (#178)
socketsupply/socket: README.md

* P2P & Cloud, built to support a new generation of apps that can connect directly to each other by providing a high performance UDP API.
* Use any backend.
* Use any frontend.
* Maintainable, zero dependencies and a smaller code base than any other competing project.
* Lean & Fast, uses a smaller memory footprint and creates smaller binaries than any other competing project.

### FAQ

Check the [Website]((https://sockets.sh/) to learn more.
 <!--we can Kyle's FAQ here, I read it yesterday and it's perfect  -->

 
### Building your first Socket app !

If you are interested in making an `Socket` app, please check [Create Socket App Repo](https://github.com/socketsupply/create-socket-app).  
`Examples` can be found in the [Examples Repo](https://github.com/socketsupply/socket-examples).  

### Documentation

The full documentation for `Socket Runtime` can be found on our [Website](https://sockets.sh/)  
The `Socket Runtime` documentation discusses API, examples , multiple guides (Apple,Desktop and Windows), P2P section and more.

### Testing

Test all the things! See the [`test/`](test/) for the unit and integration test suite.

### Contributing

> Please stop by [Discord](https://discord.com/invite/YPV32gKCsH) and ask the team about your issue and if someone is already working on it.

Reviewer: We can think about publishing install.sh and install.ps1 instructions now. These are used for building the CLI and app libs.

Author: >If you want to contribute to the Socket Runtime project itself, please connect
with any current project contributor.

I think we can have a detailed `contribution Guide` in the near future, I liked how many other websites have that. This guide can be a link in `readme` or a `detailed section` in the docs.

<p align="center">
  <img src="https://user-images.githubusercontent.com/79177582/219756869-afe51bf1-e047-4fab-9e4f-654c5bfd5393.jpeg" width="10%" height="10%"/>
</p>
<h1 align="center">Socket Runtime</h1>
                                             
<h2 align="center">⚡Write once, run anywhere, connect everyone⚡</h2>  

  
  

### Description

> `Socket Runtime`. provides a range of tools and software utilities to help developers build powerful application experiences using standard web technologies `(HTML, CSS, and JS)`.You can also connect your users, and let them communicate directly, without the cloud or any servers!  

Reviewer 2: cc @heapwolf

Reviewer 3: I think we could improve it slightly with this language:

> Web Developers use Socket Runtime to create apps for any OS, desktop or mobile. You can use plain old HTML, CSS, and JavaScript, as well as your favorite front-end libraries for example React, Svelt, and Vue.
>
> Socket Runtime exposes primitives needed for building peer-to-peer and local-first applications, such as Bluetooth, UDP, and robust file system access. Our P2P component can help you connect your app's users, and let them communicate directly, without the cloud or any servers at all.
>
> The Socket's CLI compiles applications into hybrid-native applications &mbasp; meaning, a combination of web code running in a platform's "WebView" along with platform-native code: Java on Android, Swift/Objective-C on iOS, C++ on Windows or Linux, etc.

<p align="center">
  <a href="https://github.com/socketsupply/socket"><img src="https://user-images.githubusercontent.com/79177582/219756869-afe51bf1-e047-4fab-9e4f-654c5bfd5393.jpeg" width="10%" height="10%"/>
</p>
<h1 align="center">Socket Runtime</h1>
                                             
<h2 align="center">⚡Write once, run anywhere, connect everyone⚡</h2>  

  
  

### Description

Web Developers use `Socket Runtime` to create apps for any OS, desktop, or mobile. You can use plain old HTML, CSS, and JavaScript, as well as your favorite front-end libraries for example React, Svelt, and Vue.

`Socket Runtime` exposes primitives needed for building peer-to-peer and local-first applications, such as Bluetooth, UDP, and robust file system access. Our P2P component can help you connect your app's users, and let them communicate directly, without the cloud or any servers at all.

> The Socket's CLI compiles applications into hybrid-native applications &mbasp; meaning, a combination of web code running in a platform's "WebView" along with platform-native code: Java on Android, Swift/Objective-C on iOS, C++ on Windows or Linux, etc.

Reviewer 4: ```suggestion
The Socket's CLI compiles applications into hybrid-native applications &mbasp; meaning, a combination of web code running in a platform's "WebView" along with platform-native code: Kotlin/Java on Android, Swift/Objective-C on iOS, C++ on Windows or Linux, etc.
```

Author: @chicoxyzzy for the description/intro was suggested by @heapwolf so I did the update, I'll update the readme with your changes.

Reviewer 3: > The Socket's CLI
should be `The Socket Runtime CLI`

Author: I will make the change now

### 💡 Features

* Local First, a full featured File system API & Bluetooth.
* P2P & Cloud, built to support a new generation of apps that can connect directly to each other by providing a high performance UDP API.
* Use any backend, business logic can be written in any language, Python, Rust, Node.js, etc. The backend is even completely optional.

> * Use any frontend, you can use your favorite front end framework to create your UIs, React, Svelte, Vue.

Reviewer 4: ```suggestion
* Use any frontend, you can use your favorite front end framework to create your UIs: React, Svelte, Vue.
```

* Maintainable, zero dependencies and a smaller code base than any other competing project.
* Lean & Fast, uses a smaller memory footprint and creates smaller binaries than any other competing project.

### 🔑 FAQ

Check the [Website](https://sockets.sh/) to learn more.
 <!--we can Kyle's FAQ here, I read it yesterday and it's perfect  -->

 
### 🧱 Building your first Socket app !

This is similar to React's `Create React App`, we provides a few basic boilerplates and some strong opinions so you can get coding on a production-quality app as quickly as possible.  
Please check [create-socket-app Repo](https://github.com/socketsupply/create-socket-app) to get started and to learn more.  
You can also check our `Examples` in the [Examples Repo](https://github.com/socketsupply/socket-examples).  

### 📚 Documentation

The full documentation for `Socket Runtime` can be found on our [Website](https://sockets.sh/)  

> The `Socket Runtime` documentation discusses API, examples , multiple guides(`Apple`,`Desktop`, and `Mobile`),`P2P` documentation and more.

Reviewer 4: ```suggestion
The `Socket Runtime` documentation covers Socket APIs, includes examples, multiple guides (`Apple`,`Desktop`, and `Mobile`), `P2P` documentation and more.
```

> The `Socket Runtime` documentation discusses API, examples , multiple guides (`Apple`,`Desktop`, and `Mobile`), `P2P` documentation and more.

Reviewer 4: nit: extra space before the second comma

### 🧪 Testing

> Test all the things! See the [`test/`](test/) for the unit and integration test suite.

Reviewer 4: We can add that Socket provides a built-in test runner similar to `node:test` which outputs the test results in [TAP](https://testanything.org/) format

### 🙏 Contributing

Please stop by [Discord](https://discord.com/invite/YPV32gKCsH) and ask the team about your issue and if someone is already working on it.  

> Please connect with any current project contributors : [@heapwolf][0], [@jwerle][1], or [@chicoxyzzy][2] if you want to contribute to the Socket Runtime project itself.

Reviewer 4: nit: there's an extra space before colon

# Socket Runtime
                                             

> ⚡__Write once, run anywhere, connect everyone__ ⚡

Reviewer: 🔥
